### PR TITLE
Fix `SHOW RESOURCES` DistSQL, add `customPoolProps` in result, and add unit test.

### PR DESCRIPTION
**Pull request by yx9o**
Fix `SHOW RESOURCES` DistSQL, add `customPoolProps` in result, and add unit test.

Changes proposed in this pull request:
add unit test for : DataSourceQueryResultSetTest.java

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/DataSourceQueryResultSet.java [hunk 1]
```diff
@@ -81,6 +81,7 @@ public boolean next() {
         result.put("maxPoolSize", dataSourceParameter.getMaxPoolSize());
         result.put("minPoolSize", dataSourceParameter.getMinPoolSize());
         result.put("readOnly", dataSourceParameter.isReadOnly());
+        result.put("customPoolProps", dataSourceParameter.getCustomPoolProps());
```
Comment (RaigorJiang) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/DataSourceQueryResultSet.java:84]: The result of dataSourceParameter.getCustomPoolProps() may be NULL or EMPTY, so a check is needed.

**Review by RaigorJiang — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/DataSourceQueryResultSet.java (hunk 1 above)
Comment (yx9o) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/DataSourceQueryResultSet.java]: > The result of dataSourceParameter.getCustomPoolProps() may be NULL or EMPTY, so a check is needed.

Has been updated, please review, thank you.

**Review by RaigorJiang — APPROVED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/DataSourceQueryResultSet.java (hunk 1 above)
Comment (RaigorJiang) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/rql/DataSourceQueryResultSet.java]: It's OK now.

**Review by tristaZero — APPROVED** [review]

**Comment by tristaZero** [comment]
Nice. ;-)

**Pull request merged**